src/Service.GraphQLBuilder/GraphQLUtils.cs

```diff
@@ -375,6 +376,43 @@ private static string GenerateDataSourceNameKeyFromPath(IMiddlewareContext conte
             return $"{context.Path.ToList()[0]}";
         }
 
+        /// <summary>
+        /// Helper method to determine whether a field is a column or complex (relationship) field based on its syntax kind.
+        /// If the SyntaxKind for the field is not ObjectValue and ListValue, it implies we are dealing with a column/scalar field which
+        /// has an IntValue, FloatValue, StringValue, BooleanValue, NullValue or an EnumValue.
+        /// </summary>
+        /// <param name="fieldSyntaxKind">SyntaxKind of the field.</param>
+        /// <returns>true if the field is a scalar field, else false.</returns>
+        public static bool IsScalarField(SyntaxKind fieldSyntaxKind)
+        {
+            return fieldSyntaxKind is not SyntaxKind.ObjectValue && fieldSyntaxKind is not SyntaxKind.ListValue;
```

Comment:
there's no syntaxKind or object property to condition on the affirmative? no property on `IValueNode` ?

Reply:
Whatever is not a list or an object is a scalar. I have changed the function implementation to rather use affirmatives.

On a side note, there are 8 possible SyntaxKind enum values which represent a GQL node value:

1. IntValue,
2. FloatValue,
3. StringValue,
4. BooleanValue,
5. EnumValue,
6. NullValue,
7. ListValue,
8. ObjectValue,

The last 2 represent a complex input field (relationship field).